### PR TITLE
Application eventing configuration management

### PR DESCRIPTION
**Pull request by crabtree**
**Description**

Changes proposed in this pull request:

- Implementation of mutations for managing the eventing configuration assigned for an application

**Related issue(s)**

See also #458

### CODE REVIEW

File: tests/end-to-end/director/eventing_test.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	setRuntimeLabel(t, ctx, runtime1.ID, scenariosLabel, defaultScenarios)
+	setRuntimeLabel(t, ctx, runtime1.ID, runtimeEventingURLLabelKey, runtime1EventingURL)
+
+	runtime2 := registerRuntime(t, ctx, "runtime-2-eventing")
+	defer unregisterRuntimeWithinTenant(t, runtime2.ID, defaultTenant)
+
+	setRuntimeLabel(t, ctx, runtime2.ID, scenariosLabel, defaultScenarios)
+	setRuntimeLabel(t, ctx, runtime2.ID, runtimeEventingURLLabelKey, runtime2EventingURL)
+
+	testApp := getApp(ctx, t, application.ID)
```
Comment (dbadura) [inline tests/end-to-end/director/eventing_test.go]: We can use `getApplication` from `fixtures_queries_test.go` and remove `getApp` function totally.

File: components/director/internal/domain/eventing/resolver.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+func NewResolver(transact persistence.Transactioner, eventingSvc EventingService) *Resolver {
+	return &Resolver{
+		transact:    transact,
+		eventingSvc: eventingSvc,
+	}
+}
+
+func (r *Resolver) SetDefaultEventingForApplication(ctx context.Context, app string, runtime string) (*graphql.ApplicationEventingConfiguration, error) {
+	appID, err := uuid.Parse(app)
```
[the remark this answers is not in the document]
Comment (crabtree) [reply components/director/internal/domain/eventing/resolver.go]: Maybe, I think we already have an issue with that.

File: components/director/internal/domain/eventing/resolver.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	if err = tx.Commit(); err != nil {
+		return nil, errors.Wrap(err, "while commiting the transaction")
+	}
+
+	return ApplicationEventingConfigurationToGraphQL(eventingCfg), nil
+}
+
+func (r *Resolver) UnsetDefaultEventingForApplication(ctx context.Context, app string) (*graphql.ApplicationEventingConfiguration, error) {
```
[the remark this answers is not in the document]
Comment (dbadura) [reply components/director/internal/domain/eventing/resolver.go]: I `graphql` we have `deleteDefaultEventingForApplication` mutation and in `root_resolver` we have:
```graphql
func (r *mutationResolver) DeleteDefaultEventingForApplication(ctx context.Context, appID string) (*graphql.ApplicationEventingConfiguration, error) {
	return r.eventing.UnsetDefaultEventingForApplication(ctx, appID)
}
```

File: tests/end-to-end/pkg/gql/fields.go [hunk 4]
```diff
@@ -247,3 +247,8 @@ func (fp *GqlFieldsProvider) ForPlaceholders() interface{} {
 		name
 		description`
 }
+
+func (fp *GqlFieldsProvider) ForEventingConfiguration() interface{} {
```
Comment (aszecowka) [inline tests/end-to-end/pkg/gql/fields.go]: why you return inteface{}, not string?
also, can you fix `ForPlaceholders`

File: components/director/internal/domain/eventing/resolver.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (crabtree) [reply components/director/internal/domain/eventing/resolver.go]: Well, not really, I did not want to change the public interface which was discussed in the issue describing the task.

File: components/director/internal/domain/eventing/service.go [hunk 5]
```diff
@@ -65,26 +65,92 @@ func (s *service) DeleteDefaultForApplication(ctx context.Context, appID uuid.UU
 	return model.NewApplicationEventingConfiguration(EmptyEventingURL), nil
 }
 
+func (s *service) SetForApplication(ctx context.Context, runtimeID uuid.UUID, appID uuid.UUID) (*model.ApplicationEventingConfiguration, error) {
```
Comment (aszecowka) [inline components/director/internal/domain/eventing/service.go]: `DeleteDefaultForApplication` - consider renaming, maybe `CleanupAfterUnregisteringApplication`

File: tests/end-to-end/director/eventing_test.go [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+	"fmt"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+
+	gcli "github.com/machinebox/graphql"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetDefaultRuntimeForEventingForApplication(t *testing.T) {
```
Comment (aszecowka) [inline tests/end-to-end/director/eventing_test.go]: add information why runtime1 was returned (it was the oldest one)

File: components/director/internal/domain/eventing/service.go [hunk 7]
```diff
@@ -65,26 +65,92 @@ func (s *service) DeleteDefaultForApplication(ctx context.Context, appID uuid.UU
...
+	if err != nil {
+		return nil, errors.Wrap(err, "while getting the runtime")
+	}
+
+	if !found {
+		return nil, fmt.Errorf("does not find the given runtime [ID=%s] assigned to the application scenarios", runtimeID)
+	}
+
+	if err := s.setRuntimeForAppEventing(ctx, *runtime, appID); err != nil {
+		return nil, errors.Wrap(err, "while setting the runtime as default for eveting for application")
```
[the remark this answers is not in the document]
Comment (crabtree) [reply components/director/internal/domain/eventing/service.go]: It is in the wrapped error.

**Pull request merged**